### ISSUE TITLE
How to increase speed of Detector.py? (currently it takes approx 3 hours for a 25 min video)

### ISSUE THREAD

**Issue by Franky987** [AntonMu/TrainYourOwnYOLO#122]
### System information
- **What is the top-level directory of the model you are using**: TrainYourOwnYolo/3_Inference
- **Have I written custom code (as opposed to using a stock example script provided in the repo)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux
- **TensorFlow version (use command below)**: 1.15.2
- **CUDA/cuDNN version**: 
- **GPU model and memory**: DL AMI (Ubuntu) as recommended (p2.xlarge)
- **Exact command to reproduce**: Detector.py 

### Describe the problem
From what I understood, YOLO is a relatively fast object detection method, and could be used even in real time. However, when I run the detector on a set of images or on a video, it takes a very long time to detect objects in the images/frames. As an example, a video of 25 mins at 10 FPS takes about 3 hours. How can I speed up the process? 

Note: I am not using a GPU at this time (just the CPU from my laptop). This is because uploading a video to AWS also takes a very long time. Looking forward to tips!

**Comment by bertelschmitt** [comment]
I think the best and quickest solution would be using a GPU. I get 15 - 20 FPS using a 1080 TI. To detect in real-time on 25fps footage, I simply detect only on every 2nd frame

**Comment by Franky987** [comment]
hey @bertelschmitt thanks for your response! How do you manage to realize that step in the code?

**Comment by bertelschmitt** [comment]
@Franky987:  Gladly.

It involves a little change in the source, make a backup! In def detect_video(yolo, video_path, output_path=""), which is  a routine in TrainYourOwnYOLO/2_Training/src/keras_yolo3/yolo.py , do this:
`...
 
    do_only_every = 2 # set to one to evaluate every frame, 2 for every 2nd frame (skip 1) etc 
    itercounter = 0  # run a counter to keep track
    while vid.isOpened():
        ...
        if itercounter % do_only_every  == 0: # do only every nth frame
            out_pred, image, labels = yolo.detect_image(image, show_stats=False)
        
        '''
         itercounter += 1 #at end of while loop, increment itercounter
         #if you get worried about itercounter getting too large:
        if itercounter > 1000000:   #put in whatever worries you, not getting involved in discussions about maximum integer values
            itercounter = 0  #140  #reset itercounter
`

HOWEVER, in your case, this won't be of much help.  According to your timings, you probably achieve a little over 1  frame per second on your CPU only. In order to get anywhere near real-time, you would have to send only 1 frame per second to Yolo for evaluation. which would mean  do_only_every = 10 for 10 fps video (you  need to play with the number a bit, just guessing.)

If this is o..k. for you, go for it. A lot can happen in a second ...  As I said, to get your frame rate up, you need to get a GPU

**Comment by Franky987** [comment]
Thanks a lot!! I'm going to start playing around with those numbers ;)

**Comment by bertelschmitt** [comment]
Glad to help.
